selfhosted/calibration.py

```diff
@@ -0,0 +1,46 @@
+import json
+import pathlib
+
+import click
+import yaml
```

Comment:
Same of the other one.

But Qibocal already depends on them, so it's less relevant. Nevertheless, still slightly relevant, because you could avoid versions conflict and stuffs like that (but you would have to give up on the standard format of the Qibocal runcard, that has always been YAML).

Reply:
Here, I am not sure if this is as relevant since we are importing qibocal already.

On another related note, we could provide Python API for qibocal, that allows executing routines programmatically, without having to write a YAML. This here is a potential application, but if there aren't many more, maybe it's not very useful to consider (but I think we discussed it in the past).

Reply:
> On another related note, we could provide Python API for qibocal, that allows executing routines programmatically, without having to write a YAML. This here is a potential application, but if there aren't many more, maybe it's not very useful to consider (but I think we discussed it in the past).

In a sense, Qibocal already has a Python API, if you check what the CLI is doing is just calling a single function
https://github.com/qiboteam/qibocal/blob/7abc3818fcff5eb78d9df5fe84e5454e10d60989/src/qibocal/cli/_base.py#L53

A nice (and quick) improvement could be to reshuffle the `cli` subpackage into just a `cli` top-level module, and rename the `cli` subpackage to something like `execute`/`run`/..., such that people do not have to access it as `qibocal.cli.autocalibrate` (but rather `qibocal.run.autocalibrate`).
The other option is to even avoid the subpackage and keep everything top-level (but I like grouping :P).
What do you think? @andrea-pasquale

Reply:
> In a sense, Qibocal already has a Python API, if you check what the CLI is doing is just calling a single function https://github.com/qiboteam/qibocal/blob/7abc3818fcff5eb78d9df5fe84e5454e10d60989/src/qibocal/cli/_base.py#L53

Yes, this is what I am also doing here in calibration.py, just calling `acquire` and `fit` from `qibocal.cli` seperately as I don't need the report. But this still requires defining a runcard as a dictionary (or yaml). I was thinking somehow calling directly the routine I want, eg.
```py
from qibocal.routines import single_shot_classification

single_shot_classification(platform, qubits, ...)
```
but maybe this is not very useful. Anyway, this is not very related to this PR and it's not really a priority.

Reply:
Ok, that is simple enough: my point was just that, since you're serializing a dictionary in your favorite format, you can just use JSON and avoid the further dependency.

No need to call directly single routines, this would skip all the mechanism behind the routine (i.e. storing the acquired data, fit results, ...).